app/src/main/java/pm/gnosis/heimdall/ui/safe/recover/ReplaceBrowserExtensionActivity.kt

```diff
@@ -0,0 +1,146 @@
+import pm.gnosis.utils.asEthereumAddress
+import pm.gnosis.utils.asEthereumAddressString
+import pm.gnosis.utils.nullOnThrow
+import pm.gnosis.utils.stringWithNoTrailingZeroes
+import timber.log.Timber
+import java.math.BigInteger
+
+class ReplaceBrowserExtensionActivity : ViewModelActivity<ReplaceBrowserExtensionContract>() {
+    private var submissionInProgress = false
+
+    override fun layout() = R.layout.layout_replace_browser_extension
+
+    override fun inject(component: ViewComponent) = viewComponent().inject(this)
+
+    override fun screenId() = ScreenId.REPLACE_BROWSER_EXTENSION_REVIEW
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        val txGas = nullOnThrow { intent.getStringExtra(EXTRA_TX_GAS)?.toBigInteger() } ?: run { finish(); return }
+        val dataGas = nullOnThrow { intent.getStringExtra(EXTRA_DATA_GAS)?.toBigInteger() } ?: run { finish(); return }
+        val gasPrice = nullOnThrow { intent.getStringExtra(EXTRA_GAS_PRICE)?.toBigInteger() } ?: run { finish(); return }
+        val safeTransaction = intent.getParcelableExtra<SafeTransaction>(EXTRA_SAFE_TRANSACTION) ?: run { finish(); return }
+        val signature1 =
+            nullOnThrow { intent.getStringExtra(EXTRA_ADDRESS_1).asEthereumAddress()!! to Signature.from(intent.getStringExtra(EXTRA_SIGNATURE_1)) }
+                ?: run { finish(); return }
+        val signature2 =
+            nullOnThrow { intent.getStringExtra(EXTRA_ADDRESS_2).asEthereumAddress()!! to Signature.from(intent.getStringExtra(EXTRA_SIGNATURE_2)) }
+                ?: run { finish(); return }
+
+        val chromeExtensionAddress =
+            nullOnThrow { intent.getStringExtra(EXTRA_CHROME_EXTENSION_ADDRESS).asEthereumAddress()!! } ?: run { finish(); return }
+
+        viewModel.setup(safeTransaction, signature1, signature2, txGas, dataGas, gasPrice, chromeExtensionAddress)
+
+        layout_replace_browser_extension_fee.text = "${viewModel.getMaxTransactionFee().toEther().stringWithNoTrailingZeroes()} ETH"
+    }
+
+    override fun onStart() {
+        super.onStart()
+        disposables += viewModel.loadSafe()
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribeBy(onSuccess = { safe ->
+                layout_replace_browser_extension_info_safe_image.setAddress(safe.address)
+                layout_replace_browser_extension_info_safe_address.text =
+                        safe.address.asEthereumAddressString().let { "${it.subSequence(0, 6)}...${it.subSequence(it.length - 6, it.length)}" }
+                layout_replace_browser_extension_info_safe_name.text = safe.displayName(this)
+            }, onError = Timber::e)
+
+        disposables += viewModel.observeSafeBalance()
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribeBy(onNext = ::onSafeBalance, onError = ::onSafeBalanceError)
+
+        disposables += layout_replace_browser_extension_submit.clicks()
```

Comment:
`layout_replace_browser_extension_submit` should be disabled by default

Reply:
It is in the layout 😉 

https://github.com/gnosis/safe-android/blob/a7f9297a6b1c4bcbd4d9d3aa75e731d946dbdf3b/app/src/main/res/layout/layout_replace_browser_extension.xml#L160-L170

Reply:
Added to `onStart()`